nulib/dc-nextjs
# Wireup search results using real OpenSearch results data

- Blank slate `/search`.
- Mock out Heading component.
- Update type defs for heading elements.
- Wireup Grid to results.
- Load in real results for each GridItem.
- Tidy up search methods.
- Clean house a bit more.
- Remove unused type defs.
- Remove Sticky.
- Wireup oldstyle get.
- Get search results firing a single time.
- Make queries cleaner.
- Refine type defs on search component.
- Populate search bar on router update.
- Carry searchValue when updating router.
- Finish up basic search tie ins.
- Remove useState in test.

@kdid This PR basically trims down the search page, removing facets temporarily, and integrates in the Search component as the header. To test, I setup some basic next routing that carries the query string to a URL pattern of /search?q={string}` with the `q` as a GET.  When @adamjarling is back, we'll want to dive into how we wireup storing that search param globally. 

I also have tests working locally now, so I'd like to see if this works in amplify as well.

To test:

- `npm run dev`
- browse to http://localhost:3000/
- input any string into the search bar, ex: `dog`, `joan`
- actual results  should populate in the Grid (I've disabled thumbnails)
- hit the (X) clear button in search bar, this should reset the query to default

## Notes
 - Things are expected to be unstyled right now so don't give that too much consideration.
 - All results up the limit should populate right now, paging/continuous scroll will handle breaking that out later.
 - This is still conforming to the elastic search response shape right now, we'll just need to update that at some point.
